### ISSUE TITLE
`@interval` should always return an interval

### ISSUE THREAD

**Issue by dpsanders** [dpsanders/ValidatedNumerics.jl#169]
```julia
julia> x = 1..2.1
[1, 2.10001]

julia> @interval mid(x)
1.55

julia> macroexpand(:(@interval mid(x)))
:(mid((ValidatedNumerics.convert)(Interval{parameters.precision_type},x)))

```

Maybe just wrap everything in `Interval(...)` in the `@interval` macro?
```

**Comment by lbenet** [comment]
I see your point. Yet, I think the way it works is actually the way we wanted it to behave. 

If I recall correctly, the idea was that `@interval sin(0.1)` computed the `sin` of the *correctly rounded* interval `0.1`. This should be contrasted with `Interval(sin(0.1))`, which yields a thin interval. 

So, `@interval` has to "act" inside the function, and then the function is computed. In the case of `mid`, it (re)computes the interval `x` and then computed the `mid` of `x`. So, the solution is either `Interval(mid(x))`, or
```julia
julia> xmid = mid(x);

julia> @interval(xx)
[1.54999, 1.55001]
```

**Comment by dpsanders** [comment]
My point is that `@interval mid(x)` should definitely return an interval, since that was always the desired behaviour  of `@interval`.

**Comment by lbenet** [comment]
My point is that `@interval f(x)` yields `f(@interval(x))`.

**Comment by dpsanders** [comment]
Yes, I see what is happening, but it seems to be wrong that something called interval doesn't return one!

**Comment by lbenet** [comment]
I agree with you: it's strange. However, I can't figure out how to solve this. The same will happen with `diam`, or whatever function that returns something different from an interval...

**Comment by dpsanders** [comment]
One way (I believe, without having tried) is just to modify the `@interval` macro to always wrap the result in `Interval(...)` at the end.

**Comment by dpsanders** [comment]
Fixed in #188.
